Sources/Foundation/Decimal.swift

```diff
@@ -1699,68 +1699,49 @@ public func NSDecimalString(_ dcm: UnsafePointer<Decimal>, _ locale: Any?) -> St
     if decimalSeparator == nil {
         decimalSeparator = "."
     }
+    let decimalSeparatorReversed = String(decimalSeparator.reversed())
 
-    let decimalSeparatorUtf8View = decimalSeparator.utf8
-
-    let ZERO: CChar = 0x30 // ASCII '0' == 0x30
-    let MINUS: CChar = 0x2d // ASCII '-' == 0x2d
+    let DIGITS: [String] = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"]
```

Comment:
I wouldnt use and array as it will take time to construct including allocating memory on the heap. Digit lookup could be replaced with:
```
let ZERO = UInt8(ascii: "0")

// Append '0'
str.append(String(Unicode.Scalar(ZERO)))

// Append '1'
str.append(String(Unicode.Scalar(ZERO +i )))
```

etc